docs/public-api.md

```diff
@@ -0,0 +1,65 @@
+# SwiftNIO 2 Public API
+
+The SwiftNIO project (which includes all of `github.com/apple/swift-nio*`)aims to follow [SemVer 2.0.0](https://semver.org/spec/v2.0.0.html) which requires to declare a public API. This document along with the [API documentation](https://apple.github.io/swift-nio/) declare SwiftNIO's API.
+
+### What Are Acceptable Uses of NIO's Public API
+
+- it's acceptable and expected to use any type and call any method from the `NIO*` modules unless the type or function name start with an underscore (`_`).
```

Comment:
nitpick, "it is" (as elsewhere)

Reply:
While we're here, can we capitalise the first words of all of these points? They're full sentences.

Reply:
I believe the initializers should be called out specifically. Since in case of an initializer the `_` goes to the argument label (or parameter name). For example:

```swift
struct S {
  public // not really
  init(_internalWannabe: x: Whatever) { ... }
}
```

At least these are the current rules for hiding symbols in SourceKit.